sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -67,24 +67,38 @@ def csr_mean_variance_axis0(X):
         Feature-wise variances
 
     """
-    cdef unsigned int n_samples = X.shape[0]
-    cdef unsigned int n_features = X.shape[1]
-
-    cdef np.ndarray[DOUBLE, ndim=1, mode="c"] X_data
-    X_data = np.asarray(X.data, dtype=np.float64)     # might copy!
-    cdef np.ndarray[int, ndim=1] X_indices = X.indices
+    if X.dtype == np.int32:
```

Comment:
> And is there a reason we don't support other integer sparse matrix types

I have to admit that I don't know the reason 😢 

> an int32 converted to float32 loses precision

Yeah you are right, but there's a workaround:
**Still convert `int32` and `int64` into `float64` just like before.**

What do you think?